components/RegistrationForm/index.js

```diff
@@ -11,23 +11,67 @@ import UIButtonGroupSelect from '../../UIComponents/Input/UIButtonGroupSelect';
 import { UITimeline, UITimelineEvent } from '../../UIComponents/UITimeline';
 import { getRegistration, register, uploadFile } from '../../api/api';
 import { getQueryObject } from '../../lib/path';
+import axios from 'axios';
+
+
 
 const RegistrationForm = () => {
     const [validated, setValidated] = useState(false);
     const [previousAttendance, setPreviousAttendance] = useState(true);
+    const [schoolSelected, setSchoolSelected] = useState(""); 
+    const [otherSchool, setOtherSchool] = useState(""); 
+    const [majorSelected, setMajorSelected] = useState(""); 
+    const [otherMajor, setOtherMajor] = useState(""); 
 
     let query = {};
     if (process.browser) {
         query = getQueryObject(window);
         console.log(query)
     }
 
+    function autofill(response) {
+        let firstName = document.querySelector("#firstnamefield");
+        // firstName.value = response.data.firstName
+        let lastName = document.querySelector("#lastnamefield");
+        let email = document.querySelector("#emailfield");
+        let gender = document.querySelector("#genderfield");
+        let race = document.querySelector("#firstname");
+        let graduationYear = document.querySelector("#graduationfield");
+        let school = document.querySelector("#schoolfield");
+        let major = document.querySelector("#majorfield");
+        let interest = document.querySelector("#interestfield");
+        let rpKnowledge = document.querySelector("#rpknowledgefield");
+        let attendance = document.querySelector("#attendancefield");
+    }
+
     useEffect(() => {
         if (sessionStorage.getItem('successfulRegistration') === 'true') {
-            //window.location.replace('http://localhost:3000/?registered=true')
+            // window.location.replace('http://localhost:3000/?registered=true')
             window.location.replace('https://reflectionsprojections.org/?registered=true')
+            axios.get('registration/').then(function (response) {

```

Comment:
This point in the logic is no good. The successfulRegistration token is only set after the user submits registration, this code snippet is not run prior to registration starting. You could viably set another token, or just use state, so 
`if stateGotRegistration = true
  do nothing
else
  get registration, run autofill once
`

Reply:
Also perhaps put api calls together in one file.